### DISCUSSION TITLE
Get Asset List with GraphQL query

### DISCUSSION THREAD

**Discussion by cutec-chris** [pimcore/pimcore#11630] [Q&A / Support]
Hi,

i coudnt find an way to create an query that gives me all assets stored in pimcore. 
the sample of getAssetListing in documentation returns only how much assets or asset folders are there, no ids or properties.
has someone an example how to get at least the asset ids ?

**Comment by theo-serenade** [comment]
You can query the properties using inline fragments. There are also a few different arguments you can provide, see the schema. I'm trying to `filter` on `metadata` but haven't figured it out yet.
```{
  getAssetListing(first: 3, after: 1) {
    edges {
      node {
        ... on asset {
          id
          filename
          fullpath
        }
        ... on asset_folder {
          id
          filename
          fullpath
        }
        __typename
      }
      cursor
    }
    __typename
  }
}
```
